### PR TITLE
PHP-based rest testing

### PR DESCRIPTION
**Pull request by linawolf**
depends on #427

### CODE REVIEW

**Comment by linawolf** [comment]
depends on #427

**Review by garvinhicking — APPROVED** [review]
So, now my two cents about this.

In render-guides we have made an abomination that supports all:

* Makefile
* Local PHP
* Docker / DDEV

In that repo, the "single source of truth" is a Makefile, that can accept parameters that allow each command to be executed either locally or via docker.

Then, a composer.json "make" wrapper allows to even run composer scripts calling the makefile (routing it locally, without docker).

That makes render-guides work very adapted, but it's complication hell and I do NOT want this in any of our documentation repositories, that are more "user-facing" than render-guides.

**TL;DR: We IMO have these options:**

A: Move to a "single source of truth", preferring the existing docker-way. That means, introduce Makefile, remove composer scripts.
--> Low hanging fruits. I pick this. This is also what the PR does.

B: Move to a "single source of truth", removing docker and migrating all tests and execution to be performed locally.
--> So much work to de-tangle runTests and this would IMO put us back too much.

C: Keep the duplication, allow composer AND local use
--> Too much maintenance effort.

D: Introduce the render-guides complexity to satisfy every possibility
--> I could offer to port this, but as mentioned above, this is not worth it IMO.
